ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+                            if (value_end != nullptr)
+                            {
+                                *value_end =
+                                    '\0';  // Terminate the string at newline character if found
+                            }
+                            else
+                            {
+                                value_end = strstr(value_start, "\r");
+                                if (value_end != nullptr)
+                                {
+                                    *value_end =
+                                        '\0';  // Terminate the string at newline character if found
+                                }
+                            }
+                        }
+
+                        value = strdup_function(value_start);
+                        fclose_function(file);
+                        free_function(section_header);  // Free allocated memory
+                        return value;
+                    }
+                }
+            }
+            else
+            {
+                value = section_header;  // Freed out
+                fclose_function(file);
+                return value;
+            }
+
+            // Move to the next line
+            while (start < end && *start != '\n')
+                start++;
+            start++;
+        }
+    }
+
+    if (section_header)
+        free_function(section_header);  // Free allocated memory
+    fclose_function(file);
+    return value;
+}
+
+// Retrieves EPSG codes from a CSV file based on provided geodetic identifiers.
+int MMReturnCodeFromMM_m_idofic(char *pMMSRS_or_pSRS, char *szResult,
+                                MM_BYTE direction)
+{
+    static char aEPSGCodeSRS[MM_MAX_ID_SNY];
+    char *aMMIDDBFFile = nullptr;  //m_idofic.dbf
+    FILE *pfMMSRS;
+    size_t nLong;
+    int nLongBuffer = 5000;
+    char *pszBuffer = calloc_function(nLongBuffer);
+    char *id_geodes, *psidgeodes, *epsg;
+
+    if (!pszBuffer)
+    {
+        MMCPLError(CE_Failure, CPLE_OutOfMemory,
+                   "Memory error in MiraMon "
+                   "driver (MMReturnCodeFromMM_m_idofic())");
+        return 1;
+    }
+    if (!pMMSRS_or_pSRS)
+    {
+        free_function(pszBuffer);
+        return 1;
+    }
+    memset(aEPSGCodeSRS, '\0', sizeof(*aEPSGCodeSRS));
+
+#ifdef GDAL_COMPILATION
+    aMMIDDBFFile = strdup_function(CPLFindFile("gdal", "MM_m_idofic.csv"));
+#else
+    {
+        char temp_file[MM_CPL_PATH_BUF_SIZE];
+        MuntaPath(DirectoriPrograma, strcpy(temp_file, "m_idofic.csv"), TRUE);
+        aMMIDDBFFile = strdup_function(temp_file);
+    }
+#endif
+
+    if (!aMMIDDBFFile)
+    {
+        free_function(pszBuffer);
+        MMCPLError(CE_Failure, CPLE_OpenFailed,
+                   "Error opening data\\MM_m_idofic.csv.\n");
+        return 1;
+    }
+
+    // Opening the file with SRS information
+    if (nullptr == (pfMMSRS = fopen(aMMIDDBFFile, "r")))
```

Comment:
can't you use the abstracted I/O functions like fopen_function() instead and use CPLReadLine2L() then instead of fgets(). Using fopen() could cause issues on Windows if the GDAL installation path in a directory whose name is not in the current ANSI code page

Reply:
Adopted. Using CPLReadLine2L().
By the way, there is no info about the third argument, is it?

/* -------------------------------------------------------------------- */
/*      Read a line from a text file, and strip of CR/LF.               */
/* -------------------------------------------------------------------- */
char CPL_DLL *CPLFGets(char *, int, FILE *);
const char CPL_DLL *CPLReadLine(FILE *);
const char CPL_DLL *CPLReadLineL(VSILFILE *);
**const char CPL_DLL *CPLReadLine2L(VSILFILE *, int, CSLConstList);**
const char CPL_DLL *CPLReadLine3L(VSILFILE *, int, int *, CSLConstList);

Reply:
cf https://gdal.org/api/cpl.html#_CPPv413CPLReadLine2LP8VSILFILEi12CSLConstList. It is unused / set to NULL

Reply:
Oh, ok, I was expecting to find the doc in the own .cpp/.h
Thanks